### PR TITLE
RIA-480 & RIA-600 Build case (appeal argument)

### PR DESCRIPTION
**Pull request by fantasticjamieburns**

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/DocumentWithDescription.java [hunk 1]
```diff
@@ -17,16 +18,19 @@ public DocumentWithDescription(
         Document document,
         String description
     ) {
-        this.document = document;
-        this.description = description;
+        requireNonNull(document);
+        requireNonNull(description);
+
+        this.document = Optional.ofNullable(document);
+        this.description = Optional.ofNullable(description);
```
Comment (t80) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/DocumentWithDescription.java]: Are the `.ofNullable` required when we've run them through `requireNonNull()`?

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/DocumentWithDescription.java [hunk 2]
```diff
@@ -17,16 +18,19 @@ public DocumentWithDescription(
...
     }
 
-    public Document getDocument() {
+    public Optional<Document> getDocument() {
         requireNonNull(document);
```
Comment (t80) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/DocumentWithDescription.java]: is `requireNonNull` needed? Its set to `Optional` instance at the field level so can it ever be null?

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/DocumentWithDescription.java (hunk 1 above)
Comment (fantasticjamieburns) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/DocumentWithDescription.java]: Shameful display! Not they are not required.

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/DocumentWithDescription.java (hunk 2 above)
Comment (fantasticjamieburns) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/DocumentWithDescription.java]: Deserializers can bypass constructor so yes they can still be `null`

**Pull request merged**